pkg/suite/yaml_suite.go

```diff
@@ -101,6 +102,24 @@ func convertNodes(nodeConfs map[string]YAMLNodeConf) []runtime.Node {
 func convertYAMLSuiteConfToTestCases(conf YAMLSuiteConf, fileName string) []runtime.TestCase {
 	var tests []runtime.TestCase
 	for _, t := range conf.Tests {
+		stdout := t.Stdout.(runtime.ExpectedOut)
```

Comment:
This should be moved outside of the YAML convertion for several reasons.

 - Adding a new file format would need to expand the files too, resulting in duplicate code (i.e. adding json)
 - Separation of concerns would break here because the reading of the file is considered logic of the matcher and not of the YAML convertion
 
I can understand the approach by just expanding the file and using a `TextMatcher` under the hood. I would suggest introducing a `FileMatcher` which does the same. 
It reads the file and Matches it with the `TextMatcher`. Further it would allow easier testing of the `FileMatcher` with unit tests.

Reply:
Gotcha, that makes sense. I wasn't sure where ya'll would want the file I/O to happen since it always makes testing harder.